### ISSUE TITLE
[KK] - Timeline does not inherit Eyes/Mouth,L of blendshapes when changing characters.

### ISSUE THREAD

**Issue by leoftm** [IllusionMods/HSPlugins#31]
Timeline does not inherit Eyes/Mouth,L of blendshapes when changing characters.

After loading a charastudio scene with blendshapes, some blendshapes are not inherited when changing the character.
The Eyes/Mouth,L key is present immediately after loading the scene and changing the character, but disappears the moment the timeline is moved.
Not sure if this is due to Timeline or KKPE, sorry if it is not a Timeline Issue.

The scene used for the sample is
https://discord.com/channels/447114928785063977/447116302096662528/1029445874146885753
(Note: In the video, the "Eyes/Mouth,L" key on the timeline is selected for clarity, but the problem also occurs when nothing is selected.)

https://user-images.githubusercontent.com/39472846/197362299-92253a6c-4372-4871-a9d9-d2e00bd0f81e.mp4

..Sorry for not writing the issue in the correct place.

**Comment by BullOnMars** [comment]
Yeah, having the same issue, in Honey Select 2

**Comment by BullOnMars** [comment]
It seems that this happens when you change character to another one with a different head model. for example: head mods.

**Comment by leoftm** [comment]
The information in your reply above allowed me to work around the glitch situation! Thank you very much.
This glitch seems to be the same specification as the disappearance of the skirt bones when changing clothes, or the disappearance of the animation when a body with BP bones is temporarily changed to a non-BP body.
I would like to see the specification changed to a reserved state where the animation timeline is not forced to disappear even if the bones are not present, but return when the animation timeline is restored.

**Comment by leoftm** [comment]
I feel that the phenomenon of NodesConstraints being destroyed when changing from a timeline character containing BetterPenetration to a character with a NotBP body is due to the same reason.
I don't know if this is the right place to put this request, but I would like to be able to change the headmod as well as the bodymod in charastudio.
I want to change the Headmod later like changing the bodymod to BP.

**Comment by ManlyMarco** [comment]
Looks like there is nothing in the source that can handle disappearing objects, the interpolable is just presumed dead and removed. Because of the way things are implemented it's not going to help to not remove them because they will never be updated to point to the new bone objects.

I'll see if it can be fixed in a reasonable time. If not, it'll have to wait for someone to have time for it again.

**Comment by ManlyMarco** [comment]
Actually, #37 seems to be related to this issue. Can anyone test if this build with fix from #37 works any better? [beta poseeditor builds.zip](https://github.com/IllusionMods/HSPlugins/files/10934307/beta.poseeditor.builds.zip) Use only the version for your game, backup and replace your current dll inside the plugins folder.

**Comment by leoftm** [comment]
I'm sorry, is simply replacing the BepInEx\plugins\KKPE.dll not effective?
https://cdn.discordapp.com/attachments/447116302096662528/1029445870875320331/2022_1012_0228_27_874.png
Just replacing the scene here (character was "modified headmod") to another character (was "type01") as a test did not work.
"In the meantime it looks like saving to single file and then loading after replacing might help avoid the issue,"
I am a DeepL user so I could not get a good read on the meaning of this statement.

**Comment by ManlyMarco** [comment]
Yes, replacing KKPE.dll is enough. The fix is specifically for loading and saving expression presets but it looked like it might also affect timeline. Sounds like it didn't, sadly.

**Comment by BullOnMars** [comment]
I tested again with newest plugins, still, if you change character with a different head, blend shape in Timeline will be erased/lost upon clicking play button

**Comment by leoftm** [comment]
>"In the meantime it looks like saving to single file and then loading after replacing might help avoid the issue, "

Understood! Thank you.
Load the scene data and keep the animation stopped.
Then, immediately after the headmod changes the changed character, save immediately, without touching the timeline or any other window.
If you reload that scene, the animation of the changed character in the new head will remain in place.
Thanks to this information, I was able to tentatively work around the glitch.

**Comment by leoftm** [comment]
Glitch's workaround was not universal, sorry...
If more than one person contains the "Eyes/Mouth,L" animation, the "Eyes/Mouth,L" animation may disappear even after repeatedly saving and loading the file each time you change one person at a time.

**Comment by confusionkd** [comment]
> 
> In the meantime it looks like saving to single file and then loading after replacing might help avoid the issue, though I can't test it since I don't have a way to reproduce the issue currently.

That means use timeline's "Single Files" to save the animation and then load it after changing the character.
![s](https://github.com/IllusionMods/HSPlugins/assets/114286744/0b1c8fc6-126b-453f-ace0-49f0075978b9)

**Comment by BullOnMars** [comment]
> > In the meantime it looks like saving to single file and then loading after replacing might help avoid the issue, though I can't test it since I don't have a way to reproduce the issue currently.
> 
> That means use timeline's "Single Files" to save the animation and then load it after changing the character. ![s](https://user-images.githubusercontent.com/114286744/249019991-0b1c8fc6-126b-453f-ace0-49f0075978b9.PNG)

Mostly this doesn't work, causing exceptions.

**Comment by BullOnMars** [comment]
This issue can be closed for now imo.
